### PR TITLE
#24167 Document<->Row conversion for FirestoreIO

### LINKED ISSUE #24167: [Feature Request]: Row conversion support for FirestoreIO
### What would you like to happen?

FirestoreIO should have a similar functionality like DatastoreIO does with EntityToRow/RowToEntity as that is still a missing "link" between the traditional beam schema/row and the firestore-related types.

### Issue Priority

Priority: 2

### Issue Component

Component: io-java-gcp

### PR DESCRIPTION
**Pull request by nbali**
FirestoreIO version of https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/EntityToRow.java and https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/RowToEntity.java to avoid manual mapping of objects to the Firestore native format.

Fixes #24167

TODO:
- [ ] unit tests (check if there is any case not covered)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - ~[ ] Update `CHANGES.md` with noteworthy changes.~
 - ~[ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).~

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/RowToDocument.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      MapValue.Builder mapValueBuilder = MapValue.newBuilder();
+      for (Schema.Field field : schema.getFields()) {
+        Value val = mapObjectToValue(row.getValue(field.getName()));
+        mapValueBuilder.putFields(field.getName(), val);
+      }
+      return mapValueBuilder.build();
+    }
+
+    /**
+     * Create a random key for a {@code Row} without a keyField or use a user-specified key by
```
Comment (nbali) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/RowToDocument.java:119]: outdates javadoc / missing feature TBD

**Comment by nbali** [comment]
R: @pcostell

**Comment by vishinde** [comment]
hi @nbali can you pls share when will this feature be available in master to use? I would like to use/test this functionality

**Comment by nbali** [comment]
@vishinde Quite frankly it's not up to me at this point.

**Comment by SoerenHenning** [comment]
Would be great if this could be put back on the agenda again. As far as I understand, this will also solve the issue raised here: https://stackoverflow.com/questions/71163275/add-document-to-firestore-from-beam-with-auto-generated-id

**Comment by Luiscri** [comment]
Why was this not considered? It's a great addition to the current Java SDK which will make easier to work with Firestore service

**Comment by nbali** [comment]
idk

**Pull request closed without merging**